src/diart/models.py

```diff
@@ -168,8 +206,55 @@ def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__(PyannoteLoader(model_info, hf_token))
 
     def forward(
-        self,
-        waveform: torch.Tensor,
-        weights: Optional[torch.Tensor] = None,
+            self,
+            waveform: torch.Tensor,
+            weights: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
         return self.model(waveform, weights=weights)
+
+
+class WeSpeakerSpeakerEmbeddingInference(LazyWeSpeakerSpeakerEmbedding):
+    """Minimal interface for a we speaker embedding inference."""
+
+    def __init__(self, loader: Callable[[], WeSpeakerPretrainedSpeakerEmbedding]):
+        super().__init__(loader)
+
+    @staticmethod
+    def from_pyannote(inference,
+                      ) -> "WeSpeakerSpeakerEmbeddingInference":
+        """
+        Returns an `EmbeddingModel` wrapping a pyannote model.
+
+        Parameters
+        ----------
+        inference: pyannote.audio.pipelines.speaker_verification.WeSpeakerPretrainedSpeakerEmbedding
+            The pyannote.audio inference to fetch.
+
+        Returns
+        -------
+        wrapper: EmbeddingModel
+        """
+        assert _has_pyannote, "No pyannote.audio installation found"
+        return PyannoteWeSpeakerSpeakerEmbeddingInference(inference)
+
+    def eval(self):
+        """Mock method to match pytorch api"""
+        return
+
+
+class PyannoteWeSpeakerSpeakerEmbeddingInference(WeSpeakerSpeakerEmbeddingInference):
+    def __init__(self, wespeaker_info):
+        super().__init__(PyannoteWeSpeakerSpeakerEmbeddingLoader(wespeaker_info))
+
+    def __call__(
+            self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
+    ) -> torch.Tensor:
+        self.load()
+        # Normalize weights
+        weights -= weights.min(dim=1, keepdim=True).values
+        weights /= weights.max(dim=1, keepdim=True).values
```

Comment:
You forgot to divide by `max - min` here. I suggest:

```suggestion
        min_values = weights.min(dim=1, keepdim=True).values
        max_values = weights.max(dim=1, keepdim=True).values
        weights = (weights - min_values) / (max_values - min_values)
```

Reply:
I modified them inplace, e.g. I shifted to 0 using -= and normalized them afterwards. As such, the max-min should already be happening if I'm not mistaken.

Reply:
the problem is you're only dividing by `max_values`, and you should be dividing by `max_values - min_values`

Reply:
Closing this since you've fixed it.